src/metadata-parser.h

```diff
@@ -321,6 +322,52 @@ namespace librealsense
         }
     };
 
+    template<class S, class Attribute, typename Flag>
+    class md_attribute_parser_with_crc : public md_attribute_parser<S, Attribute, Flag>
+    {
+    public: 
+        md_attribute_parser_with_crc(Attribute S::* attribute_name, Flag flag, unsigned long long offset, attrib_modifyer mod, unsigned long long crc_offset)
+            : md_attribute_parser<S, Attribute, Flag>(attribute_name, flag, offset, mod), _crc_offset(crc_offset) {}
+
+    protected:
+        unsigned long long _crc_offset; // offset of the crc inside the S class
+        bool is_attribute_valid(const S* s) const override
+        {
+            if (!md_attribute_parser<S, Attribute, Flag>::is_attribute_valid(s))
+                return false;
+
+            if (!is_crc_valid(s))
+            {
+                LOG_DEBUG("Metadata CRC mismatch");
+                return false;
+            }
+
+            return true;
+        }
+
+    private:
+        md_attribute_parser_with_crc() = delete;
+        md_attribute_parser_with_crc(const md_attribute_parser_with_crc&) = delete;
+
+        bool is_crc_valid(const S* md_info) const
+        {
+            Attribute crc = *(Attribute*)(reinterpret_cast<const uint8_t*>(md_info) + _crc_offset); //Attribute = CRC's type (ex. uint32_t)
+            auto computed_crc32 = rsutils::number::calc_crc32(reinterpret_cast<const uint8_t*>(md_info),
+                sizeof(S) - sizeof(crc));
```

Comment:
What is this for?
`sizeof(S) - sizeof(crc));`?

Reply:
S is the struct that have all of the attributes, including the crc. 
So we pass the size of all other attributes as a buffer for calc_crc32